nilearn/glm/tests/test_second_level.py

```diff
@@ -844,415 +837,384 @@ def test_second_level_residuals():
-def test_non_parametric_inference_contrast_formula(second_level_contrast):
-    with InTemporaryDirectory():
-        func_img, _ = fake_fmri_data()
-        Y = [func_img] * 4
-        rng = np.random.RandomState(42)
-        X = pd.DataFrame(rng.uniform(size=(4, 2)), columns=["r1", "r2"])
-
+    # check that passing no explicit contrast when the design
+    # matrix has more than one columns raises an error
+    rng = np.random.RandomState(42)
+    X = pd.DataFrame(rng.uniform(size=(4, 2)), columns=["r1", "r2"])
+    with pytest.raises(
+        ValueError, match="No second-level contrast is specified."
+    ):
         non_parametric_inference(
             second_level_input=Y,
             design_matrix=X,
-            second_level_contrast=second_level_contrast,
+            second_level_contrast=None,
         )
 
-        del func_img, X, Y
-
 
-def test_non_parametric_inference_contrast_computation_errors():
-    with InTemporaryDirectory():
-        func_img, mask = fake_fmri_data()
-
-        # asking for contrast before model fit gives error
-        with pytest.raises(
-            TypeError, match="second_level_input must be either"
-        ):
-            non_parametric_inference(
-                second_level_input=None,
-                second_level_contrast="intercept",
-                mask=mask,
-            )
-
-        # fit model
-        Y = [func_img] * 4
-        X = pd.DataFrame([[1]] * 4, columns=["intercept"])
-
-        ncol = len(X.columns)
-        _, cnull = np.eye(ncol)[0, :], np.zeros(ncol)
-
-        # passing null contrast should give back a value error
-        with pytest.raises(
-            ValueError,
-            match=("second_level_contrast must be a list of 0s and 1s."),
-        ):
-            non_parametric_inference(
-                second_level_input=Y,
-                design_matrix=X,
-                second_level_contrast=cnull,
-                mask=mask,
-            )
-        with pytest.raises(
-            ValueError,
-            match=("second_level_contrast must be a list of 0s and 1s."),
-        ):
-            non_parametric_inference(
-                second_level_input=Y,
-                design_matrix=X,
-                second_level_contrast=[],
-                mask=mask,
-            )
-
-        # check that passing no explicit contrast when the design
-        # matrix has more than one columns raises an error
-        rng = np.random.RandomState(42)
-        X = pd.DataFrame(rng.uniform(size=(4, 2)), columns=["r1", "r2"])
-        with pytest.raises(
-            ValueError, match="No second-level contrast is specified."
-        ):
-            non_parametric_inference(
-                second_level_input=Y,
-                design_matrix=X,
-                second_level_contrast=None,
-            )
-
-        del func_img, X, Y
+def test_second_level_contrast_computation_with_memory_caching():
+    func_img, mask = fake_fmri_data()
```

Comment:
why not pass tmp_path here?

Reply:
because ctrl + F is too hard for me... 🙈 

also that may explain why some tests were still generating data in the current directoty: thanks !